recipes/neon2sse/all/test_package/test_package.c

```diff
@@ -0,0 +1,23 @@
+#include <NEON_2_SSE.h>
+#include <stdio.h>
+
+int main()
+{
+    uint64_t a = 0x0102030405060708;
+    uint64_t b = 0x0102030405060708;
+    uint64_t c;
+    uint8x8_t va = vcreate_u8(a);
+    uint8x8_t vb = vcreate_u8(b);
+    uint8x8_t vc = vmul_u8(va, vb);
+
+    vst1_u64(&c, vc);
```

Comment:
@madebr your code was giving me a warning about an incompatible warning, so I changed this to be `vst1_u64` instead of `vst1_u8`. This made the compiler happy.
I've no idea what I'm doing tbh, so please validate this.
Thanks!

```
passing argument 1 of ‘vst1_u8’ from incompatible pointer type [-Wincompatible-pointer-types]
     vst1_u8(&c, vc);
             ^~
```

Reply:
You're correct.
`vst1_u8` stores to a u8 pointer, where we want to store to a u64 integer here.
Thanks for fixing it.

Reply:
[This](https://developer.arm.com/documentation/dui0472/k/Using-NEON-Support/NEON-intrinsics-for-storing-a-single-vector-or-lane) reference says:
``` 
void  vst1_u64(__transfersize(1) uint64_t * ptr, uint64x1_t val);  
```

So at least the type of `vc` is wrong now. But the compiler is happy, maybe there is some internal casting.

Reply:
Try the following:
```c
union {
    char c[8];
    uint64_t i;
} c;
 vst1_u64(c.c, vc);
if (c.i != 0x0104091019243140) {
...
```

Reply:
Can you try my suggestion? Then it would be perfect.

Reply:
Sorry I missed it. On it now.

Reply:
Warnings are back
```
/home/eric/githubRepos/conan-center-index-eriff/recipes/intel-neon2sse/all/test_package/test_package.c:16:14: warning: passing argument 1 of ‘vst1_u64’ from incompatible pointer type [-Wincompatible-pointer-types]
     vst1_u64(&c.c, vc);
              ^~~~
In file included from /home/eric/githubRepos/conan-center-index-eriff/recipes/intel-neon2sse/all/test_package/test_package.c:1:
/home/eric/.conan/data/intel-neon2sse/cci.20210225/asd/asd/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/NEON_2_SSE.h:9734:61: note: expected ‘uint64_t *’ {aka ‘long unsigned int *’} but argument is of type ‘char (*)[8]’
 _NEON2SSE_INLINE void vst1_u64(__transfersize(1) uint64_t * ptr, uint64x1_t val)
                                                  ~~~~~~~~~~~^~~
```

Reply:
Bug indeed, replace `vst1_u64` with `vst1_u8`.

Reply:
Similar warning
```
/home/eric/githubRepos/conan-center-index-eriff/recipes/intel-neon2sse/all/test_package/test_package.c:16:13: warning: passing argument 1 of ‘vst1_u8’ from incompatible pointer type [-Wincompatible-pointer-types]
     vst1_u8(&c.c, vc);
             ^~~~
In file included from /home/eric/githubRepos/conan-center-index-eriff/recipes/intel-neon2sse/all/test_package/test_package.c:1:
/home/eric/.conan/data/intel-neon2sse/cci.20210225/asd/asd/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/NEON_2_SSE.h:9701:59: note: expected ‘uint8_t *’ {aka ‘unsigned char *’} but argument is of type ‘char (*)[8]’
 _NEON2SSE_INLINE void vst1_u8(__transfersize(8) uint8_t * ptr, uint8x8_t val)
                                                 ~~~~~~~~~~^~~
```

Reply:
Ok, I'm going to check locally.

Reply:
This fixes it for me, and gets rid of all the warnings due to the neon2sse header.
```patch
--- a/recipes/intel-neon2sse/all/test_package/test_package.c
+++ b/recipes/intel-neon2sse/all/test_package/test_package.c
@@ -1,3 +1,7 @@
+#ifdef _MSC_VER
+#pragma warning(disable: 4996)
+#endif
+
 #include <NEON_2_SSE.h>
 #include <stdio.h>

@@ -5,14 +9,17 @@ int main()
 {
     uint64_t a = 0x0102030405060708;
     uint64_t b = 0x0102030405060708;
-    uint64_t c;
     uint8x8_t va = vcreate_u8(a);
     uint8x8_t vb = vcreate_u8(b);
     uint8x8_t vc = vmul_u8(va, vb);

-    vst1_u64(&c, vc);
+    union {
+        uint64_t i;
+        char c[8];
+    } c;
+    vst1_u8(c.c, vc);

-    if (c != 0x0104091019243140) {
+    if (c.i != 0x0104091019243140) {
         fprintf(stderr, "Wrong result\n");
         return 1;
     } else {

```

Reply:
Indeed the warning goes away with this.
Is this actually necessary? I'm building on Linux
```
+#ifdef _MSC_VER
+#pragma warning(disable: 4996)
+#endif
```

Reply:
Not needed, only for cleaner logs.

Reply:
Looking at the readme at https://github.com/intel/ARM_NEON_2_x86_SSE
You can also do the following in the test package's cmake script.
```cmake
target_compile_definitions(${PROJECT_NAME} PRIVATE NEON2SSE_DISABLE_PERFORMANCE_WARNING)
```

Reply:
Perhaps make this an option?
```python
options = {
    "sse4": [True, False],
    "performance_warnings": [True, False],
}
```
and set the defines accordingly in `package_info`.

Reply:
It does some of this automatically. So not defining `USE_SSE4` doesn't always mean it is not using SSE4.
But maybe the sse4 enabled by default makes sense.
```
#ifndef USE_SSE4
#   if defined(__SSE4_2__)
#       define USE_SSE4
#   endif
#endif
```
I'm open to add a performance_warnings option disabled by default tho.

Reply:
Done. I've added both options disabled by default to maintain the default behavior.